### PR TITLE
Reworks barber chair to be a real chair + Resprites barber chair

### PR DESCRIPTION
**Pull request by Flaborized**
## About the PR

This changes /obj/stool/barber_chair into /obj/stool/chair/comfy/barber_chair , and resprites it too! This lets you unscrew it and drag it around, disassemble it, stand up on it and jump into people, spin it as a ghost, and of course, cut hair with it. Also updates all instances of the path in maps, but there's a "You shouldn't see me" /obj/stool/barber_chair left in just in case a secret z-level has some barber chairs. Those should be replaced if this is merged!

## Why's this needed?

The old chair was immobile and I think the sprite looked sorta dated, I'd prefer a movable chair because it makes barber gimmicks much easier. It was kind of weird for the barber chair to have it's _own buckle code_ instead of just being a regular chair.

### CODE REVIEW

File: code/obj/item/table_rack_parts.dm [hunk 1]
```diff
@@ -324,6 +324,16 @@ RACK PARTS
...
 
+/obj/item/furniture_parts/barberchair
+	name = "barber chair parts"
+	desc = "A collection of parts that can be used to make a barber chair. You know, for cutting hair?"
+	icon = 'icons/obj/chairs.dmi'
+	icon_state = "barberchair_parts"
+	stamina_damage = 15
+	stamina_cost = 15
+	furniture_type = /obj/stool/chair/comfy/barber_chair
+	furniture_name = "barberchair"
```
Comment (flrsh) [inline code/obj/item/table_rack_parts.dm:335]:
```suggestion
	furniture_name = "barber chair"
```

File: code/obj/item/table_rack_parts.dm [hunk 2]
```diff
@@ -324,6 +324,16 @@ RACK PARTS
 	furniture_type = /obj/stool/chair/comfy/wheelchair
 	furniture_name = "wheelchair"
 
+/obj/item/furniture_parts/barberchair
```
Comment (flrsh) [inline code/obj/item/table_rack_parts.dm:327]:
```suggestion
/obj/item/furniture_parts/barber_chair
```

File: code/obj/ToSplit/barber_shop.dm [hunk 3]
```diff
@@ -101,12 +101,17 @@
...
+
+/obj/stool/chair/comfy/barber_chair
 	name = "barber chair"
-	desc = "Chair where hair can be cut"
-	icon = 'icons/obj/barber_shop.dmi'
+	desc = "A special chair designed for haircutting. You don't feel like any other chair would be good enough, it HAS to be one like this. You don't know why."
 	icon_state = "barberchair"
 	anchored = 1
+	arm_icon_state = "arm-barber"
+	parts_type = /obj/item/furniture_parts/barberchair
```
Comment (flrsh) [inline code/obj/ToSplit/barber_shop.dm:114]:
```suggestion
	parts_type = /obj/item/furniture_parts/barber_chair
```

**Comment by Flaborized** [comment]
I made the style changes and made outlines less dark/softer 🐝

**Comment by flrsh** [comment]
Do you want a changelog entry? If so, what should it say?

**Comment by Flaborized** [comment]
doesn't need one but you can add one if you like !!

**Pull request merged**